### DISCUSSION TITLE
patterns/chain-of-responsibility/

### DISCUSSION THREAD

**Discussion by giscus** [iluwatar/java-design-patterns-vuepress-web#48] [Announcements]
# patterns/chain-of-responsibility/

Intent Avoid coupling the sender of a request to its receiver by giving more than one object a chance to handle the request. Chain the receiving objects and pass the request alo...

https://java-design-patterns.com/patterns/chain-of-responsibility/

**Comment by javahuipaoren** [comment]
where is the method getPriority?

**Reply by iluwatar** [reply]
`RequestHandler` defines the interface and the implementation can be found in `OrcSoldier`, `OrcOfficer`, and `OrcCommander`.

**Comment by trungvinhbui** [comment]
Which tool do you use to draw Class diagram?

**Reply by iluwatar** [reply]
We use [uml-reverse-mapper](https://github.com/iluwatar/uml-reverse-mapper) tool to generate the diagram automatically from the code

**Comment by moutainMan** [comment]
I think it will be more efficient if the priority is defined when building the chain of responsibility?

**Reply by iluwatar** [reply]
Sure, but I think usually this is not an issue
